eve/main.yml

```diff
@@ -2671,3 +2692,95 @@ stages:
+            echo "$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_SUFFIX"
+      - ShellCommand:
+          name: Build shell-ui container image
+          command: docker build . --tag shell-ui:v%(prop:shell_ui_version)s
+          workdir: build/shell-ui
+          haltOnFailure: true
+      - ShellCommand:
+          name: Save shell-ui container image
+          command: >
+            docker save shell-ui:v%(prop:shell_ui_version)s |
+            gzip > shell-ui.tar.gz
+      - ShellCommand:
+          <<: *copy_artifacts
+          env:
+            DEST_DIR: "artifacts/images"
+            ARTIFACTS: >-
+              shell-ui.tar.gz
+      - Upload: *upload_artifacts
+      - ShellCommand:
+          <<: *add_final_status_artifact_success
+          env:
+            <<: *_env_final_status_artifact_success
+            STEP_NAME: build-shell-ui
+      - Upload: *upload_final_status_artifact
+
+  publish:
+    worker:
+      type: local
+    steps:
+      - ShellCommand: *add_final_status_artifact_failed
+      - SetProperty: *set_premerge_artifacts_name
+      - SetProperty: *set_premerge_url
+      - TriggerStages:
+          stage_names:
+            - publish-shell-ui
+          haltOnFailure: true
+      - ShellCommand: *add_final_status_artifact_success
+      - Upload: *upload_final_status_artifact
+
+  publish-shell-ui:
+    worker: *build_worker
+    steps:
+      - ShellCommand: *wait_for_docker
+      - ShellCommand: *image_registry_login
+      - Git: *git_pull
+      - SetPropertyFromCommand: *set_version_prop
+      - ShellCommand:
+          name: Retrieve shell-ui image from pre-merge artifacts
+          command: >
+            curl -L -s -XGET -O '%(prop:premerge_artifacts_private_url)s/images/shell-ui.tar.gz'
+          haltOnFailure: true
+      - ShellCommand:
+          name: Load shell-ui image
+          command: docker load < shell-ui.tar.gz
+          haltOnFailure: true
+      - ShellCommand:
+          name: Publish shell-ui container image
+          command: |
+            if [ "$USE_PRODUCTION_REGISTRY" = true ]; then
+                PROJECT='%(secret:harbor_prod_project)s'
+                LATEST_STABLE_RELEASE=$(git tag | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1)
+                if [ "%(prop:metalk8s_version)s" = "$LATEST_STABLE_RELEASE" ]; then
+                    docker tag shell-ui:v%(prop:metalk8s_version)s shell-ui:stable
+                fi
+            else
+                PROJECT='%(secret:harbor_dev_project)s'
+                LATEST_DEV_BRANCH=$(git branch -r --list "origin/development/[0-9].*[0-9]" | cut -d/ -f3 | sort -V | tail -n 1)
+                if [ "%(prop:branch)s" = "development/$LATEST_DEV_BRANCH" ]; then
+                    docker tag shell-ui:v%(prop:metalk8s_version)s shell-ui:latest
+                fi
+                docker tag shell-ui:v%(prop:metalk8s_version)s shell-ui:v%(prop:metalk8s_version)s-%(prop:commit_short_revision)s
+            fi
```

Comment:
To me, we want to use a property instead of checking here

Like "is_latest" and "is_stable" and then we set `is_latest` in cron job for the latest development branch in the CI directly

Reply:
Hm, yeah, why not, but so we need to change it every time we have a new branch...

Reply:
Yes, but also it's more flexible; we could for example do not publish the latest dev branch if we think that this new branch is not good enough for the moment (even if it's only development)

Reply:
We can do both now. :)